Updated registers & added ESP32 support

CircuitSetup opened the pull request:
The registers between the ATM90E32 & 36 are slightly different. These were updated for the 32. 

Added support for ESP32 - I believe the only change was giving it specific ifdefs and setting the SPI mode to 3.  The pins in the example program are the ones that should be used, but they can be changed.

I also updated how the more important calibrations values are updated. This can now be done from the main program.

CircuitSetup committed:
fixed registers for ATM90E32

Registers for ATM90E32 are a bit different from the ATM90E36.  The EMM Status Registers, SoftReset, and lack of start/checksums on other registers.

CircuitSetup committed:
comment

CircuitSetup committed:
changed file names

CircuitSetup committed:
added ESP32 support & tweaked config

CircuitSetup committed:
added seperate UART files

CircuitSetup committed:
added examples

CircuitSetup committed:
added calibration values to the ATM90E32 class

This was done so calibration can be passed from the front end program without modifying the library.

CircuitSetup committed:
Create README.md

CircuitSetup committed:
Update README.md

CircuitSetup committed:
Removed UART files - updated basic example with calibration parameters

CircuitSetup committed:
Update README.md

CircuitSetup committed:
removed "SPI" from file names & "Arduino" from folder name

CircuitSetup committed:
Merge branch 'master' of https://github.com/tavenger5/ATM90E32_Arduino

CircuitSetup committed:
removed "SPI" from names

CircuitSetup committed:
removed "SPI" from library name

CircuitSetup committed:
Update ATM90E32.cpp

CircuitSetup committed:
Merge branch 'master' of https://github.com/tavenger5/ATM90E32_Arduino

CircuitSetup committed:
added EmonCMS device

For EmonCMS device setup - https://github.com/emoncms/device 
Upload to emoncms/Modules/device/data

CircuitSetup committed:
Delete circuitsetup_split-phase.json

CircuitSetup committed:
Update README.md

CircuitSetup committed:
Delete ATM90E32.cpp

CircuitSetup committed:
Delete ATM90E32.h

CircuitSetup committed:
Delete ATM90E32_Basic_SPI.ino

CircuitSetup committed:
Updated the ATM90E32 library significantly

- Replaced lineGain with much more useful PGAGain, which increases the gain on current channels. Useful if using smaller CTs (1x) or larger CTs, like 200A (4x). The default is 2x
- Added config variable for line frequency. This allows the frequency setting to be switched in the front end software. Default is 60hz.
- Fixed GetActivePower, GetApparentPower, GetPowerFactor - all of these were not taking into account the lower 16 bit register, and were not outputting the correct numbers
- Fixed GetPowerFactor - the registers for power factor sometimes goes out of bounds for unknown reasons. Corrected this by doing a check on the output, and outputting 0 instead. Normally the value should be between -1 and 1.
- Fixed GetPhaseAngle - should have been unsigned short instead of signed. The output now outputs between 0 and 360
- Added classes for GetImportReactiveEnergy GetImportApparentEnergy, GetExportReactiveEnergy
- Added check for frequency lo and hi thresholds based on lineFreq
- Added check for voltage sag based on lineFreq, and adjustment based on lineFreq and ugain

CircuitSetup committed:
Merge branch 'master' of https://github.com/tavenger5/ATM90E32_Arduino

CircuitSetup committed:
simplified and corrected some readings for metering functions

CircuitSetup committed:
turned on B channel by default

CircuitSetup committed:
Update ATM90E32.cpp

CircuitSetup committed:
offsets & phase angle calibration

- Removed calculation that would not allow some metering values to ever be negative
- Added function to calculate Power Offset values
- Tuned offset register values which will make metering values more accurate
- Calibrated phase angle for more accurate small current values

CircuitSetup committed:
Update ATM90E32.cpp

CircuitSetup committed:
Update README.md

CircuitSetup committed:
Update README.md

whatnick commented on ATM90E32.cpp:
```diff
@@ -576,39 +585,39 @@ void ATM90E32::begin()
-  CommEnergyIC(WRITE, QoffsetB, 0x0000);    // B line reactive power offset
-  CommEnergyIC(WRITE, PoffsetC, 0x0000);    // C line active power offset
-  CommEnergyIC(WRITE, QoffsetC, 0x0000);    // C line reactive power offset
+  CommEnergyIC(WRITE, PhiC, 0x0049);        // Line calibration angle
```
Let us not hard code this.

CircuitSetup committed:
Update to constructor

- moved calibration variables to begin
- added destructor

whatnick commented on ATM90E32.h:
```diff
@@ -1,131 +1,329 @@
+#define SenergyC 0x93 		// C Apparent Energy
+
+
+/* FUNDAMENTAL / HARMONIC ENERGY REGISTERS */
```
@circuitsetup possible to run an autoformatter and clean up code here ?

CircuitSetup committed:
removed hard coded register values

CircuitSetup committed:
Clean up

removed code that was making all 32 bit registers positive
removed hard coded calibration angle values

CircuitSetup committed:
Create library.json

CircuitSetup committed:
Update README.md

CircuitSetup committed:
Update README.md

CircuitSetup committed:
Update README.md

CircuitSetup committed:
Update README.md

CircuitSetup committed:
removed power factor bitwise operations

CircuitSetup committed:
small adjustments

- added SagPeakDetCfg, Sag and Voltage peak detect period and set to 20ms
- adjusted power startup valuesto be more consistent
- removed calibrationError since there is no checksum on the 32 (there is on the 36)
- register numbers in comments
